src/palimpzest/sets.py

```diff
@@ -279,26 +284,28 @@ def add_columns(self, udf: Callable,
         """
         Add new columns by specifying UDFs.
 
-        Specify UDF for computing new columns. If you need to specify different UDFs for different columns,
-        please make multiple calls to add_columns().
-
         Examples:
-            add_columns(udf=add_name_and_age, types={'name': str, 'age': int})
-            add_columns(udf=add_birthday, types={'birthday': str})
+            add_columns(
+                udf=compute_personal_greeting,
+                types=[
+                    {'name': 'greeting', 'desc': 'The greeting message', 'type': str},
+                    {'name': 'age', 'desc': 'The age of the person', 'type': int},
+                    {'name': 'full_name', 'desc': 'The name of the person', 'type': str},
+                ]
+            )
         """
         if udf is None or types is None:
             raise ValueError("udf and types must be provided for add_columns.")
-        
+
         if isinstance(depends_on, str):
             depends_on = [depends_on]
 
         updated_cols =[]
-        for col_name, col_type in types.items():
-            new_col = {}
-            new_col["type"] = col_type
-            new_col["desc"] = "New column: " + col_name
-            new_col["name"] = col_name
-            updated_cols.append(new_col)
+        for col_dict in types:
+            assert "name" in col_dict, "each type must contain a 'name' key specifying the column name"
+            assert "type" in col_dict, "each type must contain a 'type' key specifying the column type"
+            col_dict["desc"] = col_dict.get("desc", "New column: " + col_dict["name"])
```

Comment:
nit: why don't we make desc mandatory too?

Reply:
I think the main reason is because it can make some demos visibly shorter, and for some fields like `price` in the real estate example it's self-explanatory to the point of not being necessary.

I agree however that we should strongly encourage users to provide that field